### PR TITLE
Added an option to specify the travel mode prior to getting directions

### PR DESCRIPTION
**Pull request by pobudzey**
This is the first iteration of user story #10 

Things that will eventually need to change:

- Specify all three travel modes in the API request so that the user can pick the travel mode and have the associated routes displayed in real time
- Add the Concordia Shuttle as a choice of travel mode
- UI improvements (icons, etc.)

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/directions/CallbackDirectionsConfirmListener.kt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 package com.example.campusguide.directions
 
-class CallbackDirectionsConfirmListener constructor(private val callback: (String, String) -> Unit)
+class CallbackDirectionsConfirmListener constructor(private val callback: (String, String, String) -> Unit)
     : DirectionsDialogConfirmationListener {
 
-    override fun onConfirm(start: String, end: String) {
-        callback(start, end)
+    override fun onConfirm(start: String, end: String, travelMode: String) {
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/directions/CallbackDirectionsConfirmListener.kt:6]: start, end and travelMode are getting passed around a lot together, maybe it's time to create a class (ex. DirectionsOptions) to hold them together? Would make the code a bit more readable

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -4,4 +4,9 @@
     <string name="title_activity_calculator">CalculatorActivity</string>
     <string name="loy_campus">LOY</string>
     <string name="sgw_campus">SGW</string>
+    <string-array name="travel_modes">
```
Comment (felixlapierre) [inline app/src/main/res/values/strings.xml:7]: Should we have an "Any" option?

File: app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt [hunk 3]
```diff
@@ -18,13 +19,13 @@ class GetDirectionsDialogFragment constructor(private val options: DirectionsDia
 
     /**
      * Options for creating a Directions dialog.
-     * @param message Displays appropriate instructions to user
+     * @param title Displays appropriate instructions to user
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt:22]: I think a more appropriate comment would be "The title of the dialog window"

File: app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt [hunk 4]
```diff
@@ -52,11 +54,14 @@ class GetDirectionsDialogFragment constructor(private val options: DirectionsDia
                     val start = startEditText?.text.toString()
                     val end = endEditText?.text.toString()
 
-                    options.confirmationListener.onConfirm(start, end)
+                    options.confirmationListener.onConfirm(start, end, selectedTravelMode)
                 }
                 .setNegativeButton("Cancel") { dialog, _ ->
                     dialog.cancel()
                 }
+                .setSingleChoiceItems(R.array.travel_modes, -1, DialogInterface.OnClickListener { _, which ->
```
Comment (felixlapierre) [inline app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt:62]: Would it be possible to make this allow selecting multiple options? For someone who might want walking or bus, but not car

**Review by felixlapierre — CHANGES_REQUESTED** [review]
Good stuff! A few suggestions:

- I'd recommend looking at the mockup for the feature (See the Zenhub issue). The way Google Maps does it it lets you pick different modes of transport after searching for routes rather than making you pick before searching.

- It looks like we might add more options for directions, is there a way we could architecture our solution to make it possible to add options without modifying as many existing classes? (Open-closed principle) Maybe we can leverage the Decorator pattern?

**Comment by pobudzey** [comment]
Closing because the flow needs to be improved and resemble more closely the one from the mock ups
Will rework and open another PR

**Pull request closed without merging**